src/AppBundle/Controller/UserManagment/UserController.php

```diff
@@ -0,0 +1,268 @@
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Debug\Exception\ContextErrorException;
+use Symfony\Component\Form\Extension\Core\Type\ButtonType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class UserController extends Controller
+{
+    /**
+     * @Route("/users/show/all", name="showAllUser")
+     * @Security("has_role('ROLE_elder')")
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showAllUser(Request $request)
+    {
+        //TODO Handle problem with corrupt users
+        //Create search form
+        $defaultData = array();
+        $userSearchForm = $this->createFormBuilder($defaultData)
+            ->add("filterOption", ChoiceType::class, array(
+                "choices"=>array("username"=>"filterByUid", "group"=>"filterByGroup"),
+                'label'=>false,
+                'required' => false,
+                'data'=>"filterByUid"))
+            ->add("filterText", TextType::class, array(
+                "attr"=>["placeholder"=>"search"],
+                'label'=>false,
+                'required' => false))
+            ->add("send", SubmitType::class, array(
+                "label"=>"search",
+                "attr"=>["class"=>"btn btn-lg btn-primary btn-block"]))
+            ->setMethod("get")
+            ->getForm();
+
+        $filter = new Filter();
+
+        //Handel the form input
+        $userSearchForm->handleRequest($request);
+        if ($userSearchForm->isSubmitted() && $userSearchForm->isValid()) {
+            $data = $userSearchForm->getData();
+            $filter->addFilter($data["filterOption"], $data["filterText"]);
+        }
+
+        $userRepo = $this->get("data.userRepository");
+        try {
+            $users = $userRepo->getAllUsers($filter);
+        } catch (GroupNotFoundException $e) {
+            $users = [];
+            $this->addFlash("info", $e->getMessage());
+        }
+
+        return $this->render('userManagment/showAllUsers.html.twig', [
+            "peopleSearchForm" => $userSearchForm->createView(),
+            "users"=>$users,
+        ]);
+    }
+
+    /**
+     * @Route("/users/add", name="addUser")
+     * @Security("has_role('ROLE_stavo')")
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function addUser(Request $request)
+    {
+        //Create the form
+        $userRepo = $this->get("data.userRepository");
+        $staemme = ["Ambronen","Hagen von Tronje","Anduril"];
```

Comment:
Sollen die Stämme hier fest im Code stehen? Ist es möglich diese in eine Config auszulagern?

Reply:
Nein sollen sie eigentlich nicht! Ich habe das bist jetzt nur so gemacht, weil ich mir noch nicht sicher war, wie man es sonst am besten macht:
- Config-Datei
- Datenbank
- Mit Hilfe des LDAP auslesen